<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>

  <!--Define the two robot names-->
  <arg name="sarl_tb3" default="sarl_tb3"/>
  <arg name="second_tb3" default="tb3_2"/>
  <arg name="third_tb3" default="tb3_3"/>

  <!--Define initial positions of robots-->
  <arg name="sarl_tb3_x_pos" default="-1.20"/>
  <arg name="sarl_tb3_y_pos" default="10.75"/>
  <arg name="sarl_tb3_z_pos" default="0"/>

  <arg name="second_tb3_x_pos" default="-1.20"/>
  <arg name="second_tb3_y_pos" default="10.0"/>
  <arg name="second_tb3_z_pos" default="0"/>

  <arg name="third_tb3_x_pos" default="-1.54"/>
  <arg name="third_tb3_y_pos" default="10.6"/>
  <arg name="third_tb3_z_pos" default="0"/>

  <arg name="world_file" default="$(find sarl_star_ros)/worlds/pista_world.world" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(arg world_file)"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <?ignore
  <!-- First Robot (sarl_tb3) -->
  <group ns="$(arg sarl_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model)_1.urdf.xacro" />
    <param name="tf_prefix" value="sarl_tb3_tf" />

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
	<param name="use_gui" value="False"/>
	</node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="sarl_tb3_tf" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg sarl_tb3) -x $(arg sarl_tb3_x_pos) -y $(arg sarl_tb3_y_pos) -z $(arg sarl_tb3_z_pos) -param robot_description" />
  </group>

  
  <!-- Second Robot (tb3_2) --> 
  
  <group ns="$(arg second_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model)_2.urdf.xacro" />
    <param name="tf_prefix" value="tb3_2_tf" />

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
	<param name="use_gui" value="False"/>
	</node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="tb3_2_tf" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg second_tb3) -x $(arg second_tb3_x_pos) -y $(arg second_tb3_y_pos) -z $(arg second_tb3_z_pos) -param robot_description" />
  </group>
  ?>


  
  <!-- Third Robot (tb3_3) -->
  
  <group ns="$(arg third_tb3)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model)_3.urdf.xacro" />
    <param name="tf_prefix" value="tb3_3_tf" />

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
	<param name="use_gui" value="False"/>
	</node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="tb3_3_tf" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg second_tb3) -x $(arg third_tb3_x_pos) -y $(arg third_tb3_y_pos) -z $(arg third_tb3_z_pos) -param robot_description" />
  </group>
  

</launch>

